charms/istio-pilot/src/charm.py

```diff
@@ -321,7 +353,7 @@ def _get_gateway_address(self):
         """
         # FIXME: service name is hardcoded
         svcs = self.lightkube_client.get(
-            Service, name="istio-ingressgateway", namespace=self.model.name
+            Service, name="istio-ingressgateway-workload", namespace=self.model.name
```

Comment:
This is already in the latest charm code, please rebase your branch and omit this change.